extensions/functions_datetime.yaml

```diff
@@ -287,3 +362,255 @@ scalar_functions:
+            value: string
+        return: string
+      - args:
+          - name: x
+            value: date
+          - name: format
+            value: string
+        return: string
+      - args:
+          - name: x
+            value: time
+          - name: format
+            value: string
+        return: string
+  -
+    name: "round_temporal"
+    description: >-
+      Round a given timestamp/date/time to a multiple of a time unit. If the given timestamp is not already an
+      exact multiple from the origin in the given timezone, the resulting point is chosen as one of the
+      two nearest multiples. Which of these is chosen is governed by rounding: FLOOR means to use the earlier
+      one, CEIL means to use the later one, ROUND_TIE_DOWN means to choose the nearest and tie to the
+      earlier one if equidistant, ROUND_TIE_UP means to choose the nearest and tie to the later one if
+      equidistant.
+
+      Timezone strings must be as defined by IANA timezone database (https://www.iana.org/time-zones).
+      Examples: "Pacific/Marquesas", "Etc/GMT+1".
+      If timezone is invalid an error is thrown.
+    impls:
+      - args:
+          - name: x
+            value: timestamp
+          - name: rounding
+            options: [ FLOOR, CEIL, ROUND_TIE_DOWN, ROUND_TIE_UP ]
+          - name: unit
+            options: [ YEAR, MONTH, WEEK, DAY, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND ]
+          - name: multiple
+            value: i64
+          - name: origin
+            value: timestamp
+        return: timestamp
+      - args:
+          - name: x
+            value: timestamp_tz
+          - name: rounding
+            options: [ FLOOR, CEIL, ROUND_TIE_DOWN, ROUND_TIE_UP ]
+          - name: unit
+            options: [ YEAR, MONTH, WEEK, DAY, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND ]
+          - name: multiple
+            value: i64
+          - name: timezone
+            description: Timezone string from IANA tzdb.
+            value: string
+          - name: origin
+            value: timestamp_tz
+        return: timestamp_tz
+      - args:
+          - name: x
+            value: date
+          - name: rounding
+            options: [ FLOOR, CEIL, ROUND_TIE_DOWN, ROUND_TIE_UP ]
+          - name: unit
+            options: [ YEAR, MONTH, WEEK, DAY, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND ]
+          - name: multiple
+            value: i64
+          - name: origin
+            value: date
+        return: date
+      - args:
+          - name: x
+            value: time
+          - name: rounding
+            options: [ FLOOR, CEIL, ROUND_TIE_DOWN, ROUND_TIE_UP ]
+          - name: unit
+            options: [ YEAR, MONTH, WEEK, DAY, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND ]
+          - name: multiple
+            value: i64
+          - name: origin
+            value: time
+        return: time
+  -
+    name: "round_calendar"
```

Comment:
Are these descriptions identical or am I misreading?  What is the difference between `round_temporal` and `round_calendar`?

Reply:
`round_temporal` is for physical units (e.g. 24hrs) and `round_calendar` is for calendrical units (e.g. 1 day, 1 month).